extensions/wsc/src/main/java/org/mule/extension/ws/internal/connection/ClientFactory.java

```diff
@@ -174,21 +164,16 @@ private void removeInterceptor(List<Interceptor<? extends Message>> inIntercepto
     inInterceptors.removeIf(i -> i instanceof PhaseInterceptor && ((PhaseInterceptor) i).getId().equals(name));
   }
 
-  /**
-   * Tries to find the address where the web service is located or fail.
-   */
-  private String getSoapAddress(Port port) throws ConnectionException {
-    if (port != null) {
-      for (Object address : port.getExtensibilityElements()) {
-        if (address instanceof SOAPAddress) {
-          return ((SOAPAddress) address).getLocationURI();
-        } else if (address instanceof SOAP12Address) {
-          return ((SOAP12Address) address).getLocationURI();
-        } else if (address instanceof HTTPAddress) {
-          return ((HTTPAddress) address).getLocationURI();
-        }
+  private WscDispatcher createDispatcher(String address, HttpService httpService, String transportConfig)
+      throws ConnectionException {
+    String protocol = address.substring(0, address.indexOf("://"));
```

Comment:
@alegmarra parsing jms:// urls with java URL throws a not recognized protocol exception, what are you using in the jms extension?